docs/sources/upgrading/_index.md

```diff
@@ -31,6 +31,26 @@ The output is incredibly verbose as it shows the entire internal config struct u
 
 ## Main / Unreleased
 
+### Loki
+
+#### Error responses from API
+
+The body of HTTP error responses from API endpoints changed from plain text to
+JSON. The `Content-Type` header was previously already set incorrectly to
+`application/json`. Therefore returning JSON fixes this incosistency
+
+The response body has the following schema:
+
+```json
+{
+  "code": <http status code>,
+  "message": "<error message>",
+  "status": "error"
+}
+```
+
+"404 Not Found" errors still return responses with content type `text/plain`.
```

Comment:
Why do we special-case 404s? This is going to make writing Loki clients complex.

Reply:
404 errors are handled differently than errors that occur in our handlers. However, it should be possible to override the NotFoundHandler of the router.

Reply:
Can you look into that please? We should really try have consistency across the board if possible.